### PR TITLE
NClient benchmark

### PR DESCRIPTION
**Pull request by smolchanovsky**

### CODE REVIEW

**Comment by Kingmidas74** [comment]
Self review? So uncivilized %-)

**Comment by smolchanovsky** [comment]
> Self review? So uncivilized %-)

GitHub is broken, I can't merge without approval 🤪
https://github.com/github/feedback/discussions/8797

File: benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            _flurlClient = new FlurlClient(_api.Urls.First());
+            _nclient = NClientGallery.Clients.GetRest().For<INClientJsonHttpResponseClient>(_api.Urls.First()).Build();
+            _refitClient = Refit.RestService.For<IRefitJsonHttpResponseClient>(_api.Urls.First());
+            _restEaseClient = RestEase.RestClient.For<IRestEaseJsonHttpResponseClient>(_api.Urls.First());
+        }
+        
+        [Benchmark]
+        public void HttpClient_CreateAndSend()
+        {
+            var httpClient = new HttpClient();
```
Comment (Kingmidas74) [inline benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:46]: How many times httpClient will be created? I could be wrong, but multiple httpclient creation is smth bad, isn't?

File: benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            
+            _httpClient = new HttpClient();
+            _restSharpClient = new RestSharp.RestClient(_api.Urls.First());
+            _flurlClient = new FlurlClient(_api.Urls.First());
+            _nclient = NClientGallery.Clients.GetRest().For<INClientJsonHttpResponseClient>(_api.Urls.First()).Build();
+            _refitClient = Refit.RestService.For<IRefitJsonHttpResponseClient>(_api.Urls.First());
+            _restEaseClient = RestEase.RestClient.For<IRestEaseJsonHttpResponseClient>(_api.Urls.First());
+        }
+        
+        [Benchmark]
```
Comment (Kingmidas74) [inline benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:43]: What about "cold start"? Does this library take this into account by default?

**Review by Kingmidas74 — COMMENTED** [review]
Some questions for sharing of knowledge

File: benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs (hunk 1 above)
Comment (smolchanovsky) [reply benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:46]: You're right, in real applications you need to make the client a singleton, or use a factory. But in this case, the number of connections is not enough to lead to a socket exhaustion problem

File: benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs (hunk 2 above)
Comment (smolchanovsky) [reply benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:43]: Yes, warming up by default. Cold start must be specified explicitly: https://benchmarkdotnet.org/articles/guides/choosing-run-strategy.html

File: benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs (hunk 1 above)
Comment (Kingmidas74) [reply benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:46]: Ok. Any proofs? I could not find any relevant explicit settings  in files.

Comment (smolchanovsky) [reply benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:46]: This class has the attribute `[SimpleJob(invocationCount: 2000, targetCount: 10)]`

Comment (Kingmidas74) [reply benchmark/NClient.Benchmark.Client/JsonHttpResponseClient/JsonHttpResponseClientBenchmark.cs:46]: Sorry. My mistake. This is in attr, yeah?

**Review by Kingmidas74 — APPROVED** [review]
Feature request: extrude params of tests runner in settings

**Pull request merged**